### ISSUE TITLE
[BUG] Explicit interface implementation issue

### ISSUE THREAD

**Issue by ahdung** [icsharpcode/ILSpy#1682]
Version: 5.0.0.4776-preview1

![image](https://user-images.githubusercontent.com/12094247/64059942-5337bb00-cbf8-11e9-9439-c2d0cd459f4d.png)

**Comment by siegfriedpammer** [comment]
Yes, please provide the DLL. Seems like there is something missing in the metadata tables.

Thank you!

**Comment by ahdung** [comment]
I thank you, you sir are so great.

at `DevComponents.DotNetBar.PanelEx` last content.
[FastReport.Bars.zip](https://github.com/icsharpcode/ILSpy/files/3562343/FastReport.Bars.zip)

**Comment by siegfriedpammer** [comment]
Thank you very much!

It's just like I suspected: There is no info in the `Property`, `PropertyMap` and `MethodSemantics` tables for these methods. I think, the only thing we could do is strip invalid modifiers from the declarations when generating C# code.

**Comment by ahdung** [comment]
@siegfriedpammer I think you can do more thing, like check it by Interface, Ixxx has `Enabled` property, class shouldn't has Ixxx.get_Enabled().

**Comment by yyjdelete** [comment]
See an reproduce for this, and it can be generated with roslyn.
Seems roslyn will generated those proxy method for interface when
>case 1: none of ancestor impl the interface, and the class(ClassA1A) and the ancestor(ClassA2A) which the impl methods come from different assembly.

>case 2: the impl methods(ClassA1C or ClassA1D) is defined in subclass of first ancestor which impl the interface (ClassA1E) but not override(directly use newslot or override an newslot) the methods in it.

Not sure if there is any document for this, but if those generated methods is removed manually: 
For `case 1`: get an `System.TypeLoadException: Method "xxx" is not implemented` for runtime, and `Error: Class implements interface but not method, Class: '' Interface: '' Missing method` for ILVerify.
And for `case 2` matches the wrong method.

```cs
//In assembly1
    public class ClassA1A : ClassA2A, IA
    {
    }
    public class ClassA1B : ClassA1C, IA
    {
    }
    public class ClassA1C : ClassA1D
    {
        public new int A { get; set; } = 1;
    }
    public class ClassA1D : ClassA1E
    {
        public new void C() { Console.WriteLine("ClassA1D"); }
    }
    public class ClassA1E : IA
    {
        public int A { get; set; }
        public event Action B;
        public void C() { Console.WriteLine("ClassA1E"); }
    }

    //In any assembly
    public interface IA
    {
        int A { get; set; }
        event Action B;
        void C();
    }

//In another assembly2
public class ClassA2A
{
    public int A { get; set; }
    public event Action B;
    public void C() { Console.WriteLine("A2A"); }
}
```

**Comment by siegfriedpammer** [comment]
Thank you very much for providing these reproduction steps!

**Comment by siegfriedpammer** [comment]
See Roslyn source code for the rules used by the C# compiler:

https://github.com/dotnet/roslyn/blob/fb028b7c198c2f726067bb8627cb014f1154a541/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs#L1466-L1523
